src/gameStateLogger.cpp

```diff
@@ -533,7 +533,14 @@ void GameStateLogger::writeStationEntry(JSONGenerator& json, P<SpaceStation> sta
 void GameStateLogger::writeMissileEntry(JSONGenerator& json, P<MissileWeapon> missile)
 {
     json.write("category_modifier", missile->category_modifier);
-    json.write("owner_id", missile->owner->getMultiplayerId());
+
+    // A missile's owner might not exist when we log data. Skip the owner_id if so.
+    P<SpaceObject> owner = missile->owner;
+
+    if (owner)
```

Comment:
This can just be `if (missile->owner) {}` no need for the local variable. Or else use the local variable to get the ID from.